mantidproject/mantid
# Remove Spectra Algorithm

**Description of work.**
Creation of a new algorithm based around removing spectra from a workspace. Has the option to remove masked spectra, remove spectra with invalid detectors, and remove spectra at a given workspace indices list.

**To test:**
- Check tests pass

*Checking remove spectra with No detectors*
- Run this script:
```python
Load("TSC04970.raw", OutputWorkspace="TSC04970")
ConvertUnits(InputWorkspace="TSC04970", OutputWorkspace="TSC04970", Target="DeltaE", EMode="Indirect")
RemoveSpectra(InputWorkspace="TSC04970", OutputWorkspace="TSC04970_Removed_spectra", RemoveSpectraWithNoDetector=True)
```
- TSC04970_Removed_spectra should have 2 less histograms/spectra than TSC04970

*Checking remove spectra with a mask*
- Run this script:
```python
CreateSampleWorkspace(OutputWorkspace="ws")
MaskDetectors(Workspace="ws", SpectraList="1,2,3")
RemoveSpectra(InputWorkspace="ws", OutputWorkspace="ws1", RemoveMaskedSpectra=True)
```
- ws1 should have 3 less histograms/spectra than ws

*Checking remove spectra when given a list*
- Run this script:
```python
CreateSampleWorkspace(OutputWorkspace="ws2")
RemoveSpectra(InputWorkspace="ws2", OutputWorkspace="ws3", WorkspaceIndices="0,1,2")
```
- ws3 should have 3 less histograms/spectra than ws2

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
